test/e2e/wmco_test.go

```diff
@@ -131,3 +144,67 @@ func testWindowsNodeDeletion(t *testing.T) {
 	}
 	defer testCtx.Cleanup()
 }
+
+// testWMCOCustomResourceDefinitionValidations tests if validations of the fields of WindowsMachineConfigs CRD are
+// working as expected
+func testWMCOCustomResourceDefinitionValidations(t *testing.T) {
+	testCtx := framework.NewTestCtx(t)
+	namespace, err := testCtx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create WMCO custom resource without replicas field
+	wmcoWithoutReplicasField := &operator.WindowsMachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "WindowsMachineConfig",
+			APIVersion: wmcApiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "instance",
+			Namespace: namespace,
+		},
+		Spec: operator.WindowsMachineConfigSpec{
+			InstanceType: ec2InstanceType,
+			AWS:          &operator.AWS{CredentialAccountID: awsAccountID, SSHKeyPair: sshKey},
+		},
+	}
+
+	err = framework.Global.Client.Create(context.TODO(), wmcoWithoutReplicasField,
+		&framework.CleanupOptions{TestContext: testCtx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	// Failing the test if WMCO custom resource is not throwing an error when replicas field is absent
+	require.NotEmpty(t, err, "Creation of WMCO custom resource did not throw an error when required replicas "+
+		"field is absent")
+	// Failing the test if WMCO custom resource is throwing an error, but the error is not related to replicas field
+	// being absent
+	require.Contains(t, err.Error(), "spec.replicas: Required value",
```

Comment:
This should be assert as its what were testing for and not required for something else